### PR TITLE
fix: fix duplicate entries

### PR DESCRIPTION
**Pull request by slimm609**
- fix duplicate entries in fortifed count
- check binary to validate that it relies on glibc before comparing against glibc

### CODE REVIEW

**Comment by teoberi** [comment]
The solution chosen to test the libc dependency:
`ldd "${1}" 2> /dev/null | grep 'libc\.so' | cut -d' ' -f3`
differs a bit from the one I thought of, that is was:
`${readelf} -d "${1}" 2> /dev/null | grep 'NEEDED' | grep 'libc\.so'`
or (maybe)
`${readelf} -d "${1}" 2> /dev/null | grep 'NEEDED' | grep 'libc\.so' | awk '{print $5}`

Source:
https://www.baeldung.com/linux/show-shared-libraries-executables

**3. Using the ldd Command**
...
The ldd command is pretty handy to list the shared libraries of a program.
However, **we should use it with caution, as the ldd utility may execute the program to get the list of the shared libraries. We should never run the ldd command on untrusted executables.**

**7. Conclusion**
In this article, we’ve discussed different ways to list shared libraries of a program.
The ldd command is the most straightforward one to show the shared libraries of a program. The readelf command is a better choice compared to the rest since we deal with “ELF” format on Linux. However, we must keep in mind that we should never use it on untrusted executables.

**Comment by teoberi** [comment]
For your request from here: https://github.com/slimm609/checksec.sh/pull/236#discussion_r1574613537 maybe we could add to Fortify,  for "N/A" case in **tests/binaries**:
1. from [here](https://github.com/tyyu3/LiblessHelloWorld) a version of "Hello World" in C + ASM to test the binary file case **without libc dependency**
Build commands:
```
gcc -o hello main.c start.S hello.S -w -nostdlib -no-pie -s
clang -o hello_cl main.c start.S hello.S -w -nostdlib -no-pie -s
gcc -m32 -o hello32 main.c start.S hello.S -w -nostdlib -no-pie -s
clang -m32 -o hello_cl32 main.c start.S hello.S -w -nostdlib -no-pie -s
```
2. from [here](https://opensource.com/article/21/6/linux-checksec) a version of "Hello World" in C  to test the binary file case when **"${FS_cnt_total}" == "0"**
Build commands:
```
gcc -o helloworld helloworld.c -w -D_FORTIFY_SOURCE=0 -O2 -s
clang -o helloworld_cl helloworld.c -w -D_FORTIFY_SOURCE=0 -O2 -s
gcc -m32 -o helloworld32 helloworld.c -w -D_FORTIFY_SOURCE=0 -O2 -s
clang -m32 -o helloworld_cl32 helloworld.c -w -D_FORTIFY_SOURCE=0 -O2 -s
```
Source files:
[hello.zip](https://github.com/slimm609/checksec.sh/files/15129183/hello.zip)
[helloworld.zip](https://github.com/slimm609/checksec.sh/files/15129186/helloworld.zip)

File: kernel_configs/build_kernel_configs.sh [hunk 1]
```diff
@@ -19,17 +19,17 @@ build_config() {
...
   build_config 4.x 4 "$i"
 done
 
 #build configs for 5.x up to 5.10
-for i in {1..10}; do
+for i in {1..15}; do
   build_config 5.x 5 "$i"
 done
+
+#build configs for 5.x up to 5.10
```
Comment (teoberi) [inline kernel_configs/build_kernel_configs.sh:32]: Did it copy from 5?
`#build configs for 6.x up to 6.8`

Comment (slimm609) [reply kernel_configs/build_kernel_configs.sh]: updated

**Pull request merged**